### PR TITLE
[COSIM] Added thread-safe tests

### PR DESCRIPTION
**Pull request by ksco**
Also added `BOX64_DYNAREC_TEST=2` in the CI, well, except for RISC-V, need to find out what's the issue there...

### CODE REVIEW

**Comment by ksco** [comment]
Current CI failing seems have nothing to do with this PR.

**Comment by ksco** [comment]
Okay, the RISC-V side is complaining about the newly enabled X87 TAGS test, will port it to RISC-V later..

**Comment by ptitSeb** [comment]
main is green now

**Comment by ksco** [comment]
ack, rerunning

File: src/dynarec/arm64/dynarec_arm64_00.c [hunk 1]
```diff
@@ -3082,6 +3083,7 @@ uintptr_t dynarec64_00(dynarec_arm_t* dyn, uintptr_t addr, uintptr_t ip, int nin
         case 0xE9:
         case 0xEB:
             BARRIER(BARRIER_MAYBE);
+            if (box64_dynarec_test == 2) { NOTEST(x1); }
```
Comment (ptitSeb) [inline src/dynarec/arm64/dynarec_arm64_00.c:3086]: mmm, NOTEST on JMP defeat the purpose of thest. You should NOTEST only if the jump is outside the block.

Comment (ksco) [reply src/dynarec/arm64/dynarec_arm64_00.c:3086]: the issue is I can't know if the jump is outside or not from the interpreter..

File: src/dynarec/arm64/dynarec_arm64_0f.c [hunk 2]
```diff
@@ -1551,6 +1551,7 @@ uintptr_t dynarec64_0F(dynarec_arm_t* dyn, uintptr_t addr, uintptr_t ip, int nin
             break;
 
         #define GO(GETFLAGS, NO, YES, F)   \
+            if (box64_dynarec_test == 2) { NOTEST(x1); }                \
```
Comment (ptitSeb) [inline src/dynarec/arm64/dynarec_arm64_0f.c:1554]: same

File: src/dynarec/la64/dynarec_la64_00.c [hunk 3]
```diff
@@ -511,6 +511,7 @@ uintptr_t dynarec64_00(dynarec_la64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
             break;
 
         #define GO(GETFLAGS, NO, YES, F, I)                                                         \
+            if (box64_dynarec_test == 2) { NOTEST(x1); }                                            \
```
Comment (ptitSeb) [inline src/dynarec/la64/dynarec_la64_00.c:514]: same

File: src/dynarec/la64/dynarec_la64_00.c [hunk 4]
```diff
@@ -1601,6 +1602,7 @@ uintptr_t dynarec64_00(dynarec_la64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
         case 0xE9:
         case 0xEB:
             BARRIER(BARRIER_MAYBE);
+            if (box64_dynarec_test == 2) { NOTEST(x1); }
```
Comment (ptitSeb) [inline src/dynarec/la64/dynarec_la64_00.c:1605]: same

File: src/dynarec/arm64/dynarec_arm64_00.c (hunk 1 above)
Comment (ptitSeb) [reply src/dynarec/arm64/dynarec_arm64_00.c:3086]: All this seems very complex, and with a lot of function signature changes. if NOTEST(x1) only does runtime changes, that's fine.

Comment (ksco) [reply src/dynarec/arm64/dynarec_arm64_00.c:3086]: Yeah, that’s what it takes :(

**Comment by ptitSeb** [comment]
I'll merge, but I really think that disabling test on jump, espcialy conditionnal jump, greatly reduce the usefullness of the test. Flags are not tested (because of the deffered and on-demand nature of the them), only their side effect, like conditionnal jump taken or not, are observed. Disbling jump blinds you from most flag computation error...

**Comment by ksco** [comment]
> I'll merge, but I really think that disabling test on jump, espcialy conditionnal jump, greatly reduce the usefullness of the test. Flags are not tested (because of the deffered and on-demand nature of the them), only their side effect, like conditionnal jump taken or not, are observed. Disbling jump blinds you from most flag computation error...

Sorry I missed this yesterday, I agree, the limitations are too much. But think it as an enhancement for `DYNAREC_TEST=1`, not a replacement -- it's nice to have a more limited but working tool for multi-thread programs.

**Pull request merged**